src/apps/detect-projects/default.nix

```diff
@@ -0,0 +1,23 @@
+{
+  pkgs,
+  utils,
+  apps,
+  ...
+}:
+utils.writePureShellScriptBin
+"detect-projects"
+(with pkgs; [
+  coreutils
+  yq
+])
+''
+  set -e
+
+  # Use $1 as source if set. Otherwise default to ./.
+  export source=$(realpath ''${1:-.})
+
+  ${apps.callNixWithD2N} eval --json \
+    "dream2nix.functions.discoverers.discoverProjects2 {source = builtins.getEnv \"source\";}" \
+    | yq --toml-output \
+    | (cat ${./template.toml} && echo "" && cat)
+''
```

Comment:
This doesn't need to be addressed right now but I think it could be cool if perhaps this would let users choose translator and builder in an easy fashion so that they wouldn't need to edit `projects.toml` directly. It could present the builders and translators as options and show a small description and link to more extensive documentation in case (it could also be useful to include documentation links in the generated `projects.toml` just so that it is more accessible).

Reply:
Where would you suggest to have these options, if not in the project.toml?
The `project.toml` suits well for editing, why not have it there?

In general I like the idea. We could have commented out options, similar to the generated `configuration.nix` in nixos.
But one significant difference is, that we might output several projects withing one `project.toml`, so I'm not sure if inline explanations of fields will bloat the whole thing too much.
Maybe have a big comment at the beginning of the file, explaining all the options and referencing the documentation?

Reply:
I meant options as in a CLI app, when you run it it would let you choose the builder / translator etc. For documentation we can just add it at the start of `projects.toml` as links like `builder: https://nix-community.github.io/dream2nix/subsystems/foo/bar`.

Reply:
> For documentation we can just add it at the start of projects.toml as links like builder: https://nix-community.github.io/dream2nix/subsystems/foo/bar.

This would result in dead links for several subsystems as of now. Maybe we should first ensure that we have a complete option reference generated via our nixos modules?
@antotocar34 Already got us a step further via https://github.com/nix-community/dream2nix/pull/399, but there are still some parts left to complete, like:
https://github.com/nix-community/dream2nix/blob/3d6f13ef9d4cb8c41bc83383bbec3e74865ef90d/src/modules/flake-parts/makeOutputsArgs.nix#L18-L23